### PR TITLE
Backport CP-43755 - rate limits for PAM logins

### PR DESCRIPTION
**Pull request by lindig**
We are backporting a list of commits from master but drop the changes to the datamodel and instead use a hard-coded value.

### CODE REVIEW

File: ocaml/xapi/xapi_session.ml [hunk 1]
```diff
@@ -26,17 +26,17 @@ open Client
 open Auth_signature
 open Extauth
 
+let with_lock = Xapi_stdext_threads.Threadext.Mutex.execute
+
```
Comment (psafont) [inline ocaml/xapi/xapi_session.ml:30]: This binding isn't needed, all the users were changed to semaphores

**Review by psafont — APPROVED** [review]
I'm surprised that the location of the vendored semaphore location works, but the backport seems solid

**Comment by psafont** [comment]
The vendored module can't be located, easiest thing is probably to move it to ocaml/xapi

**Comment by lindig** [comment]
Sorry - I pushed to the wrong branch. This needs still reviewing

**Pull request merged**